src/api/Startup.cs

```diff
@@ -63,7 +63,8 @@ public void ConfigureServices(IServiceCollection services)
                     options.SerializerSettings.PreserveReferencesHandling = Newtonsoft.Json.PreserveReferencesHandling.Objects;
                 }
             );
-            services.AddScoped<ICourseDbContext>(provider => provider.GetService<CourseDbContext>());
+            var context = services.BuildServiceProvider().GetService<CourseDbContext>();
+            services.AddSingleton<ICourseDbContext>(provider => context);

```

Comment:
@timabell its probably this line we need to change to scope instead of singleton

services.AddSingleton<ILocationRequester, LocationRequester>();

Reply:
ta
singleton would be re-used across all requests which risks cross-talk with different user's requests. I'll keep looking at this